docs/general/ddl/data-types.rst

```diff
@@ -303,1283 +458,2517 @@ Casting from ``OBJECT`` to ``JSON``::
-``precision`` of a numeric is the total count of significant digits in the
-unscaled numeric value.  The ``scale`` of a numeric is the count of decimal
-digits in the fractional part, to the right of the decimal point. For example,
-the number 123.45 has a precision of 5 and a scale of 2. Integers have a scale
-of zero.
+::
 
-To declare the ``numeric`` type with the precision and scale use the syntax::
+    cr> INSERT INTO my_table (
+    ...     number
+    ... ) VALUES (
+    ...     32767
+    ... );
+    CREATE OK, 1 row affected (... sec)
 
-    NUMERIC(precision, scale)
+.. HIDE:
 
-Alternatively, only the precision can be specified, the scale will be zero
-or positive integer in this case::
+    cr> REFRESH TABLE my_table;
+    REFRESH OK, 1 row affected (... sec)
 
-    NUMERIC(precision)
+::
 
-Without configuring the precision and scale the ``numeric`` type value will be
-represented by an unscaled value of the unlimited precision::
+    cr> SELECT number FROM my_table;
+    +--------+
+    | number |
+    +--------+
+    | 32767  |
+    +--------+
+    SELECT 1 row in set (... sec)
 
-    NUMERIC
+.. HIDE:
 
-The ``numeric`` type backed internally by the Java ``BigDecimal`` class. For
-more detailed information about its behaviour, see `BigDecimal documentation`_.
+    cr> DROP TABLE my_table;
+    DROP OK, 1 row affected (... sec)
+
+
+.. _type-integer:
 
-.. _ip-type:
+``INTEGER``
+'''''''''''
 
-``ip``
-======
+An integer.
 
-The ``ip`` type allows to store IPv4 and IPv6 addresses by inserting their
-string representation. Internally IP addresses are stored as ``bigint``
-allowing expected sorting, filtering and aggregation.
+Limited to four bytes, with a range from -2^31 to 2^31-1.
 
 Example::
 
-    cr> create table my_table_ips (
-    ...   fqdn text,
-    ...   ip_addr ip
+    cr> CREATE TABLE my_table (
+    ...     number INTEGER
     ... );
     CREATE OK, 1 row affected (... sec)
 
 ::
 
-    cr> insert into my_table_ips (fqdn, ip_addr)
-    ... values ('localhost', '127.0.0.1'),
-    ...        ('router.local', '0:0:0:0:0:ffff:c0a8:64');
-    INSERT OK, 2 rows affected (... sec)
-
-::
-
-    cr> insert into my_table_ips (fqdn, ip_addr)
-    ... values ('localhost', 'not.a.real.ip');
-    SQLParseException[Cannot cast `'not.a.real.ip'` of type `text` to type `ip`]
-
-IP addresses support the :ref:`operator <gloss-operator>` ``<<``, which checks
-for subnet inclusion using `CIDR notation`_. The left-hand :ref:`operand
-<gloss-operand>` must be of type :ref:`ip <ip-type>` and the right-hand must be
-of type :ref:`text <data-type-text>` (e.g., ``'192.168.1.5' <<
-'192.168.1/24'``).
-
-.. _date-time-types:
-
-Date and time types
-===================
+    cr> INSERT INTO my_table (
+    ...     number
+    ... ) VALUES (
+    ...     2147483647
+    ... );
+    CREATE OK, 1 row affected (... sec)
 
-+---------------------------------+----------+-------------------------+------------------------+
```

Comment:
Minor, i think i quite liked the overview table of the various time related types.

Reply:
The "overview" is now apparently rendered through a "table of contents", see [1]. Is this okay for you? Otherwise, I will also be happy to bring back the table as seen at [2]. Now that you are outlining it to me, I also like it ;].

[1] https://crate--11631.org.readthedocs.build/en/11631/general/ddl/data-types.html#dates-and-times
[2] https://crate.io/docs/crate/reference/en/4.6/general/ddl/data-types.html#date-and-time-types

Reply:
Hi Michael,

I shuffled some ideas back and forth and finally settled to bring back a whole section dedicated to listing _all_ supported data types in tabular form, along with some related facts, with 571c03805d.

You can preview it at [1]. What do you think about it?

With kind regards,
Andreas.

[1] https://crate--11631.org.readthedocs.build/en/11631/general/ddl/data-types.html#data-types-overview

Reply:
Hi @seut,

this is a minor detail which is not finished yet because I wanted to hear about your feedback.

With kind regards,
Andreas.